services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/monitor/core.py

```diff
@@ -0,0 +1,412 @@
+        )
+
+        while self._keep_running:
+            # make sure access to the dict is locked while the monitoring cycle is running
+            try:
+                async with self._lock:
+                    await self._runner()
+
+                await sleep(settings.monitor_interval_seconds)
+            except asyncio.CancelledError:
+                break
+
+        logger.warning("Monitor was shut down")
+
+    async def start(self) -> None:
+        # run as a background task
+        logging.info("Starting dynamic-sidecar monitor")
+        self._keep_running = True
+        asyncio.get_event_loop().create_task(self._run_monitor_task())
+
+        # discover all services which were started before and add them to the monitor
+        dynamic_sidecar_settings: DynamicSidecarSettings = (
+            self._app.state.settings.dynamic_services.dynamic_sidecar
+        )
+        services_to_monitor: Deque[
+            ServiceLabelsStoredData
+        ] = await get_dynamic_sidecars_to_monitor(dynamic_sidecar_settings)
+
+        logging.info(
+            "The following services need to be monitored: %s", services_to_monitor
+        )
+
+        for service_to_monitor in services_to_monitor:
+            (
+                service_name,
+                node_uuid,
+                service_key,
+                service_tag,
+                paths_mapping,
+                compose_spec,
+                container_http_entry,
+                dynamic_sidecar_network_name,
+                simcore_traefik_zone,
+                service_port,
+                project_id,
+                user_id,
+            ) = service_to_monitor
+
+            monitor_data = MonitorData.assemble(
+                service_name=service_name,
+                node_uuid=node_uuid,
+                project_id=project_id,
+                user_id=user_id,
+                hostname=service_name,
+                port=dynamic_sidecar_settings.port,
+                service_key=service_key,
+                service_tag=service_tag,
+                paths_mapping=paths_mapping,
+                compose_spec=compose_spec,
+                container_http_entry=container_http_entry,
+                dynamic_sidecar_network_name=dynamic_sidecar_network_name,
+                simcore_traefik_zone=simcore_traefik_zone,
+                service_port=service_port,
+            )
+            await self.add_service_to_monitor(monitor_data)
+
+    async def shutdown(self):
+        logging.info("Shutting down dynamic-sidecar monitor")
+        self._keep_running = False
+        self._inverse_search_mapping = dict()
+        self._to_monitor = dict()
+
+
+def _get_monitor(app: FastAPI) -> DynamicSidecarsMonitor:
+    return app.state.dynamic_sidecar_monitor
+
+
+def get_monitor(request: Request) -> DynamicSidecarsMonitor:
+    return _get_monitor(request.app)
+
+
+async def setup_monitor(app: FastAPI):
+    dynamic_sidecars_monitor = DynamicSidecarsMonitor(app)
```

Comment:
you might rename the file? what means core.py? it's a background task.
what about monitor_task.py ?

Reply:
Thanks, I've tried a few times to rename it but was not able to find an appropriate name 🤣